### PR TITLE
Add section about exclusive params with expressions, and emphasize command output lines

### LINKED ISSUE #162: Example for using expressions with exclusive parameters
If you use exclusive parameters (as documented in the [Advanced Parameters](https://www.commonwl.org/user_guide/11-records/index.html)) section together with expressions you might end up with a situation where an expression is undefined. An example of such non-working CWL is [here](https://gist.github.com/pvanheus/f6bdc3ead51eacbee7b6d59398d41f39).

The solution is to use logical OR to provide a default value to the expression (thanks to @tetron) for explaining this. I.e. expressions like `$(inputs.output.output_file || [])`. This works because expressions are [ECMAScript expressions](https://www.commonwl.org/v1.0/CommandLineTool.html#Expressions), something I didn't realise until last night.

I am preparing a pull request to add this explanation to the User Guide but I cannot right now come up with a simple example illustrating this, so I'm opening this issue to ask if anyone can come up with a straightforward example and also to remind myself to complete this addition to the guide.

### PR DESCRIPTION
**Pull request by kinow**
Closes #162 

Emphasizes command output lines of interest for users reading the text.

Renames “Advanced Inputs” to “Inclusive and Exclusive Inputs”. I thought it made sense as that section is specifically about these two variations of arguments. I realized a better name could be useful after reading the TODO notes, where we have future content for this section that could be considered advanced too, but I think it would be best to use separate sections. What others think? 

Adds a new section below the “Inclusive and Exclusive Inputs”, called ”Exclusive Input Parameters with Expressions”, using an example similar to what is described in #162.

### CODE REVIEW

**Comment by kinow** [comment]
The first commit updates the existing docs to emphasize the lines of interest in the command output. See the before and after below:

1/

image
image

2/

image
image

3/

image
image

I did that because I didn't see the warning in the second code block, where the important part is **“record-job2.yml:6:3: Warning: invalid field `itemD`, expected one of: 'itemC'”**.

**Comment by kinow** [comment]
Suggestion to rename the Advanced Inputs section to Include and Exclusive Inputs

image

Rationale: In the future we might add new sub-sections to the Inputs section. Some could be considered Advanced Inputs too?

image

**Comment by kinow** [comment]
Added new section about exclusive params with expressions, preview available here: https://common-workflow-languageuser-guide--247.org.readthedocs.build/en/247/topics/inputs.html#exclusive-input-parameters-with-expressions

File: src/conf.py [hunk 1]
```diff
@@ -101,7 +101,7 @@
     '08-arguments/index.md': '../topics/additional-arguments-and-parameters.html',
     '09-array-inputs/index.md': '../topics/inputs.html',
     '10-array-outputs/index.md': '../topics/outputs.html',
-    '11-records/index.md': '../topics/inputs.html',
+    'inputs/index.md': '../topics/inputs.html',
```
Comment (mr-c) [inline src/conf.py:104]: ? I don't understand.

**Review by mr-c — CHANGES_REQUESTED** [review]
In general, I quite like this; thanks!

File: src/topics/inputs.md [hunk 2]
```diff
@@ -200,108 +200,177 @@ This will be demonstrated in the next lesson
...
 :language: yaml
 :caption: "`record-job1.yml`"
 :name: record-job1.yml
 ```
 
 ```{code-block} console
+:emphasize-lines: 6-7
 $ cwl-runner record.cwl record-job1.yml
-Workflow error, try again with --debug for more information:
+INFO /home/kinow/Development/python/workspace/user_guide/venv/bin/cwl-runner 3.1.20220830195442
```
[the remark this answers is not in the document]
Comment (kinow) [reply src/topics/inputs.md:225]: Yup. I have an idea to use sphinx + docker to automate it, but haven't started workijg on it yet. Unfortunately there are paths in most of the other sections/pages. Can I postpone it for a follow up and try to tackle all paths in the user guide at once?

File: src/conf.py (hunk 1 above)
Comment (kinow) [reply src/conf.py:104]: huh. that's odd. I don't remember changing that for this PR. Will revert it, sorry.

Comment (kinow) [reply src/conf.py:104]: Ah, the IDE refactor did that change. Reverted.

File: src/topics/inputs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kinow) [reply src/topics/inputs.md:225]: Issue where I started working on this (will start working on it again now): https://github.com/common-workflow-language/user_guide/issues/148

**Review by mr-c — APPROVED** [review]

**Pull request merged**